internal/master/service.go

```diff
@@ -42,18 +44,53 @@ type standaloneService struct {
 	lg         *zap.Logger
 	statsCli   stats.Client
 	regionInfo *serverpb.RegionInfo
+	isClosed bool
```

Comment:
probably `isActive`  might make more sense than this. Also that will avoid the negative check at all places later. Also wouldn't a `shutdown` chan maybe better here instead of just a bool?

Reply:
1. Have used the same convention as the one used for other implementations. 
2. Have kept this boolean flag as an identifier in case the called of a closed service. Have also introduced a 1 buffer channel for closing events.